pylint/checkers/variables.py

```diff
@@ -1103,6 +1117,55 @@ def __init__(self, linter: PyLinter) -> None:
         """This is a queue, last in first out."""
         self._postponed_evaluation_enabled = False
 
+    @utils.only_required_for_messages(
+        "unbalanced-dict-unpacking",
+    )
+    def visit_for(self, node: nodes.For) -> None:
+        if not isinstance(node.target, nodes.Tuple):
+            return
+
+        targets = node.target.elts
+
+        inferred = utils.safe_infer(node.iter)
+        if inferred is None or not isinstance(inferred, DICT_TYPES):
```

Comment:
```suggestion
        if not isinstance(inferred, DICT_TYPES):
```

Reply:
oh good point.... None is never going to be an instance of DICT_TYPES :D